probe-rs/src/flashing/flash_algorithm.rs

```diff
@@ -216,12 +217,13 @@ impl FlashAlgorithm {
 
         let assembled_instructions = raw.instructions.chunks_exact(size_of::<u32>());
 
-        if !assembled_instructions.remainder().is_empty() {
-            return Err(FlashError::InvalidFlashAlgorithmLength {
-                name: raw.name.to_string(),
-                algorithm_source: Some(target.source().clone()),
-            });
-        }
+        let last_iter = if !assembled_instructions.remainder().is_empty() {
+            let remainder = assembled_instructions.remainder();
+            let word = u32::from_le_bytes(remainder.iter().cloned().chain(repeat(0u8)).take(4).collect::<Vec<_>>().try_into().unwrap());
+            repeat(word).take(1)
```

Comment:
I'd love to understand what's going on here but honestly I have no idea 😅

Reply:
I basically take the last remainder and pad it with zeros.

The repeat(word).take(1) ensure that both branches have the same type.

Reply:
I'd prefer `Some(last word)` and `None`, can that work?

Reply:
Also are you looking for `[0u8;4]` with that `repeat(0u8)).take(4).collect::<Vec<_>>().try_into().unwrap()` monster? :D

Reply:
Yes it can. Put it there.